062-support-tiered-storage.md

```diff
@@ -0,0 +1,131 @@
+    # A map with String keys and String values. Keys will be automatically prefixed with `rsm.config.`.
+
+```
+
+### RemoteStorageManager (RSM)
+
+The RSM config only supports custom type. The config allows users to specify the RSM class information.
+
+Because the `RemoteStorageManager` dependency is provided by a Kafka plugin, the implementation could vary case by case. It’s important to allow users to pass in additional configuration values within the supplied prefix range. The custom configuration can be supplied via  `spec.kafka.config` with the predefined prefix.
+
+### RemoteLogMetadataManager (RLMM)
+
+For simplicity, the proposal doesn't add customization for `RemoteLogMetadataManager`. If tiered storage is enabled, the default implementation using `TopicBasedRemoteLogMetadataManager` will be used, and corresponding configuration will be appended.
+
+The Strimzi Cluster Operator will automatically configure any required configurations for the RLMM such as the basic thread pool configs required to create the internal Kafka client and talk to the brokers.
+
+Any additional RLMM configuration options such as performance tuning or security config options can be put in `.spec.kafka.config` directly using the `rlmm.config.` prefix.
+
+Example can be found below.
+
+### Example configuration
+
+The below config define a sample configuration for tiered storage setup, using a `custom` RSM type.
+```
+kafka:
+  tieredStorage:
+    type: customRemoteStorageManager
+    className: com.example.kafka.tiered.storage.s3.S3RemoteStorageManager
+    classPath: /opt/kafka/plugins/tiered-storage-s3/*
+    config:
+    # A map with String keys and String values. Key fields will be automatically prefixed with `rsm.config.` and appended to Kafka broker config.
+      storage.bucket.name: my-bucket
+  config:
+    ...
+    rlmm.config.remote.log.metadata.common.client.bootstrap.servers: broker-bootstrap.com:9091
+    rlmm.config.remote.log.metadata.common.client.security.protocol: SSL
+    rlmm.config.remote.log.metadata.topic.replication.factor: 1
+```
+
+The configuration above will get translated to the below Kafka broker config:
+```
+# RLMM configuration generated by Strimzi
+remote.log.storage.system.enable: true
+remote.log.metadata.manager.impl.prefix: rlmm.config.
+remote.log.metadata.manager.class.name: org.apache.kafka.server.log.remote.metadata.storage.TopicBasedRemoteLogMetadataManager
+#RSM configs set by the operator and by the user
+remote.log.storage.manager.class.name: com.example.kafka.tiered.storage.s3.S3RemoteStorageManager
+remote.log.storage.manager.class.path: /opt/kafka/plugins/tiered-storage-s3/*
+remote.log.storage.manager.impl.prefix: rsm.config.
+rsm.config.storage.bucket.name=my-bucket
+# # Custom RLMM configs
+rlmm.config.remote.log.metadata.common.client.bootstrap.servers: broker-bootstrap.com:9091
+rlmm.config.remote.log.metadata.common.client.security.protocol: SSL
+rlmm.config.remote.log.metadata.topic.replication.factor: 1
```

Comment:
As described in the comment earlier, I think this iwll likely look more like this:

```suggestion
# RLMM configuration generated by Strimzi
remote.log.storage.system.enable=true
remote.log.metadata.manager.class.name=org.apache.kafka.server.log.remote.metadata.storage.TopicBasedRemoteLogMetadataManager
remote.log.metadata.manager.impl.prefix=rlmm.config.
remote.log.metadata.manager.listener.name=REPLICATION-9091
rlmm.config.remote.log.metadata.common.client.security.protocol=SSL
rlmm.config.remote.log.metadata.common.client.ssl.keystore.location=/tmp/kafka/cluster.keystore.p12
rlmm.config.remote.log.metadata.common.client.ssl.keystore.password=${CERTS_STORE_PASSWORD}
rlmm.config.remote.log.metadata.common.client.ssl.keystore.type=PKCS12
rlmm.config.remote.log.metadata.common.client.ssl.truststore.location=/tmp/kafka/cluster.truststore.p12
rlmm.config.remote.log.metadata.common.client.ssl.keystore.password=${CERTS_STORE_PASSWORD}
rlmm.config.remote.log.metadata.common.client.ssl.truststore.type=PKCS12

# Custom RLMM configs
rlmm.config.remote.log.metadata.topic.replication.factor=1

# RSM configs set by the operator and by the user
remote.log.storage.manager.class.path=/opt/kafka/plugins/tiered-storage-s3/*
remote.log.storage.manager.class.name=com.example.kafka.tiered.storage.s3.S3RemoteStorageManager
remote.log.storage.manager.impl.prefix=rsm.config.
rsm.config.storage.bucket.name=my-bucket
```

Reply:
@scholzj, the config I added is slightly different than the one you mentioned. I replied before but to make it clear, those `rlmm.config.remote.log.metadata.common.client` config should not fall into Strimzi generated config. It needs client customization because the Kafka cluster could use different SSL mechanism and it doesn't necessarily use these config. So I put these into `# RSM configs set by the operator and by the user`

Reply:
I'm not sure I understand/agree. It needs to connect to the Kafka cluster. That can be done through the replication listener po port 9091 which is used for other internal tooling and has its security configured by the operator. So it is (currently) always mTLS. So this can be configured by the operator and should be configured by the operator.

Reply:
You are right. Updated. Thanks for the catch!